src/test/scala/io/citrine/lolo/util/LoloPyDataLoaderTest.scala

```diff
@@ -0,0 +1,111 @@
+package io.citrine.lolo.util
+import io.citrine.lolo.TestUtils
+import io.citrine.lolo.learners.RandomForest
+import io.citrine.lolo.stats.functions.Friedman
+import org.junit.Test
+
+class LoloPyDataLoaderTest {
+
+  private def getData: Array[Byte] = {
+    val stream = getClass.getClassLoader.getResourceAsStream("4by4double.bytes")
+    val byteArray = new Array[Byte](128)
+    stream.read(byteArray)
+    byteArray
+  }
+
+  @Test
+  def testFeatureArray(): Unit = {
+    val data = LoloPyDataLoader.getFeatureArray(getData, 4, bigEndian = false)
+    assert(data.length == 4)
+    assert(data.head.length == 4)
+    assert(Math.abs(data.head(1) - 1) < 1e-6, "Array read in wrong order")
+  }
+
+  @Test
+  def test1DArrayDouble(): Unit = {
+    val data = LoloPyDataLoader.get1DArray(getData, getFloat = true, bigEndian = false).asInstanceOf[Seq[Double]]
+    assert(data.length == 16)
+    assert(Math.abs(data(1) - 1) < 1e-6, "Array read badly")
+  }
+
+  @Test
+  def test1DArrayInteger(): Unit = {
+    val data = LoloPyDataLoader.get1DArray(getData, getFloat = false, bigEndian = false).asInstanceOf[Seq[Int]]
+    assert(data.length == 32)
+    assert(data.take(3).sum == 0) // First 3 values are all 0
+    assert(data(3) == 1072693248) // Fourth is a big one
+  }
+
+  @Test
+  def testZipping(): Unit = {
+    val X = LoloPyDataLoader.getFeatureArray(getData, 4, bigEndian = false)
+    val y = LoloPyDataLoader.get1DArray(getData, getFloat = true, bigEndian = false).asInstanceOf[Seq[Double]]
+    val trainData = LoloPyDataLoader.zipTrainingData(X, y)
+    assert(trainData.length == 4)
+    val (x_i, y_i: Double) = trainData.head
+    assert(x_i.length == 4)
+    assert(Math.abs(x_i(1) - 1) < 1e-6)
+    assert(Math.abs(y_i - 0) < 1e-6)
+  }
+
+  @Test
+  def testGetRegression(): Unit = {
+    // Generate an RF model
+    val trainingData = TestUtils.binTrainingData(
+      TestUtils.generateTrainingData(32, 12, noise = 0.1, function = Friedman.friedmanSilverman),
+      inputBins = Seq((0, 8))
+    )
+
+    val RFMeta = RandomForest.apply(numTrees = trainingData.length * 2)
+      .train(trainingData)
+    val RF = RFMeta.getModel()
+
+    // Generate the prediction results object
+    val results = RF.transform(trainingData.map(_._1))
```

Comment:
Why do the results need to come from a model?  Couldn't you just generate some random means and sigmas and then round-robin them through?

Reply:
Yes.  You can create an anonymous class that extends `PredictionResult` (https://stackoverflow.com/questions/24657450/scala-how-to-use-extends-with-an-anonymous-class)

Reply:
Cool. I've done that for both of the test where I trained a model